File: mybatis-plus-join-test/test-join/src/main/java/com/github/yulichang/test/join/dto/UserDTO.java
package com.github.yulichang.test.join.dto;

import com.github.yulichang.test.join.entity.UserDO;
import com.github.yulichang.test.join.enums.Sex;
import lombok.Data;
import lombok.ToString;

import java.util.List;
import java.util.Map;


/**
 * userDTO
 */
@Data
@ToString
public class UserDTO {
    /** user */
    private Integer id;
    /** user */
    private Map<String,String> name;
    /** user */
    private Sex sex;
    /** user */
    private String headImg;
    /** user */
    private String userHeadImg;//同 headImg 别名测试
    /** user_address */
    private String tel;
    /** user_address */
    private String address;
    /** user_address */
    private String userAddress;
    /** area */
    private String province;
    /** area */
    private String city;
    /** area */
    private Map<String, String> area;

    private List<AddressDTO> addressList;

    private List<UserDO> children;
}

File: mybatis-plus-join-test/test-kotlin/src/main/java/com/github/yulichang/test/kt/dto/UserDTO.kt
package com.github.yulichang.test.kt.dto

import com.github.yulichang.test.kt.entity.UserDO
import com.github.yulichang.test.kt.enums.Sex
import lombok.Data
import lombok.ToString

/**
 * userDTO
 */
@Suppress("unused")
@Data
@ToString
class UserDTO {
    /** user  */
     val id: Int? = null

    /** user  */
     val PName: String? = null

    /** user  */
     val json: Map<String, String>? = null

    /** user  */
     val sex: Sex? = null

    /** user  */
     val headImg: String? = null

    /** user  */
     val userHeadImg: String? = null //同 headImg 别名测试

    /** user_address  */
     val tel: String? = null

    /** user_address  */
     val address: String? = null

    /** user_address  */
     val userAddress: String? = null

    /** area  */
     val province: String? = null

    /** area  */
     val city: String? = null

    /** area  */
     val area: Map<String, String>? = null
     val addressList: List<AddressDTO>? = null
     val addressDTO: AddressDTO? = null
     val addressIds: List<Int>? = null
     val children: List<UserDO>? = null
}
